src/mono/wasm/runtime/startup.ts

```diff
@@ -747,3 +753,17 @@ export type DownloadAssetsContext = {
     loaded_files: { url?: string, file: string }[],
     loaded_assets: { [id: string]: [VoidPtr, number] },
 }
+
+/// Called when dotnet.worker.js receives an emscripten "load" event from the main thread.
+///
+/// Notes:
+/// 1. Emscripten skips a lot of initialization on the pthread workers, Module may not have everything you expect.
+/// 2. Emscripten does not run the preInit or preRun functions in the workers.
+/// 3. At the point when this executes there is no pthread assigned to the worker yet.
+async function mono_wasm_pthread_worker_init(): Promise<void> {
+    // This is a good place to attach listeners for the pthread MessagePorts using pthreads_worker.addThreadCreatedCallback();
+    console.debug("mono_wasm_pthread_worker_init");
+    pthreads_worker.addThreadCreatedCallback((thread_id) => {
```

Comment:
This is confusing me, it seems that this is called on the worker, right ?
Bu the callback collection is not shared across workers ?

Reply:
every time dotnet.js is loaded, the callback array is initialized.  if you load dotnet.js in two different webworkers, you will get two different arrays.  there is no shared memory (except SharedArrayBuffer) in WebWorkers.  each one is completley isolated.